### PR TITLE
Integrate Facility-Clearing-UI-Indicator fix into CHL

### LINKED ISSUE #1181: Integrate Facility Clearing UI Indicator Fix
There's a very small bug fixed by this standalone mod: https://steamcommunity.com/sharedfiles/filedetails/?id=1130714554

I think it makes sense to fix it in Highlander to remove the need for MCO and to cull another mod from people's list.

### PR DESCRIPTION
**Pull request by BlackDog86**
Fixes #1181 

Added UIFacilityGrid_FacilityOverlay & replaced appropriate code with that from the Clearing-UI-Indicator fix.

Tested & working as expected - engineer staffed in second excavation slot appears in the second slot on the UI.

### CODE REVIEW

**Comment by BlackDog86** [comment]
Anything else needed on this one?

**Review by Iridar — CHANGES_REQUESTED** [review]

**Comment by Iridar** [comment]
> Anything else needed on this one?

Apparently forgot to click the button to finish the review and my messages weren't visible to you.

**Comment by Iridar** [comment]
Still has unaddressed issues since the last review.

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/UIFacilityGrid_FacilityOverlay.uc [hunk 1]
```diff
@@ -838,15 +838,51 @@ simulated function UpdateGridUI(TUIRoomData kData)
...
+	Tuple.Data[4].kind = XComLWTVFloat; // Label height
+	Tuple.Data[4].f = MC.GetNum("queueLabel.label.textHeight");
+	Tuple.Data[5].kind = XComLWTVBool; // Label multiline
+	Tuple.Data[5].b = false;
+	Tuple.Data[6].kind = XComLWTVBool; // Label word wrap
+	Tuple.Data[6].b = false;
+	Tuple.Data[7].kind = XComLWTVBool; // Label autosize
+	Tuple.Data[7].b = false;
+
+	`XEVENTMGR.TriggerEvent('FacilityUICustomizeHook', Tuple, Room);
```
[the remark this answers is not in the document]
Comment (BlackDog86) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/UIFacilityGrid_FacilityOverlay.uc:873]: Ah those are just hooks from the original mod to allow over-riding some facility status labels. Not important at all, will remove.

**Comment by BlackDog86** [comment]
Updated & tested. Should be ready for merge.

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/UIFacilityGrid_FacilityOverlay.uc (hunk 1 above)
[the remark this answers is not in the document]
Comment (RustyDios) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/UIFacilityGrid_FacilityOverlay.uc]: If you are integrating the mod, then surely NOT including these hooks could break any mods dependant on them from the original mod ??

[the remark this answers is not in the document]
Comment (Iridar) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/UIFacilityGrid_FacilityOverlay.uc]: We're integrating the bugfix, not the mod.

**Review by Iridar — APPROVED** [review]
Looks sane, gonna merge after testing.

**Comment by Iridar** [comment]
Tested, works fine. Good work.

**Pull request merged**